subsys/bluetooth/host/hci_core.c

```diff
@@ -3095,6 +3095,8 @@ static int start_le_scan(u8_t scan_type, u16_t interval, u16_t window)
 			}
 
 			set_param.addr_type = BT_ADDR_LE_RANDOM;
+		} else {
+			set_random_address(&bt_dev.id_addr[0].a);
```

Comment:
This will only do the right thing if the identity address is a random address. If it's a public address this would try to write that as a random address. I think instead of `else` you want `else if (set_param.addr_type == BT_ADDR_LE_RANDOM)`

Reply:
yes, I will fix it